test/test_grid_addboundaryset.jl

```diff
@@ -0,0 +1,202 @@
+function extractboundarycheck(grid::Ferrite.AbstractGrid{3})
+    faces = _extractboundarycheck(grid, addfaceset!, grid.facesets)
+    edges = _extractboundarycheck(grid, addedgeset!, grid.edgesets)
+    vertices = _extractboundarycheck(grid, addvertexset!, grid.vertexsets)
+    return union(faces, edges, vertices)
+end
+function extractboundarycheck(grid::Ferrite.AbstractGrid{2})
+    faces = _extractboundarycheck(grid, addfaceset!, grid.facesets)
+    vertices = _extractboundarycheck(grid, addvertexset!, grid.vertexsets)
+    return union(faces, vertices)
+end
+@testset "grid getentities" begin
+#                            (8)
+#                   (7) +-----+-----+(9)
+#                       |\    |\    |
+#                       | \ 6 | \ 8 |
+#                       |  \  |  \  |
+#                       | 5 \ | 7 \ |
+#                       |    \|    \|
+#                   (4) +-----+-----+(6)
+#                       |\    |\    |
+#                       | \ 2 | \ 4 |
+#                       |  \  |  \  |
+#                       | 1 \ | 3 \ |
+#                       |    \|    \|
+#                   (1) +-----+-----+(3)
+#                            (2)
+    grid = generate_grid(Triangle, (2, 2));
+    topology = ExclusiveTopology(grid);
+    for cell in 1:8
+        @test Ferrite.getfacevertices(grid, FaceIndex(cell, 1)) == Set([VertexIndex(cell, 1), VertexIndex(cell, 2)])
+        @test Ferrite.getfacevertices(grid, FaceIndex(cell, 2)) == Set([VertexIndex(cell, 2), VertexIndex(cell, 3)])
+        @test Ferrite.getfacevertices(grid, FaceIndex(cell, 3)) == Set([VertexIndex(cell, 3), VertexIndex(cell, 1)])
+    end
+end
+@testset "grid getinstances" begin
+#                            (8)
+#                   (7) +-----+-----+(9)
+#                       |\    |\    |
+#                       | \ 6 | \ 8 |
+#                       |  \  |  \  |
+#                       | 5 \ | 7 \ |
+#                       |    \|    \|
+#                   (4) +-----+-----+(6)
+#                       |\    |\    |
+#                       | \ 2 | \ 4 |
+#                       |  \  |  \  |
+#                       | 1 \ | 3 \ |
+#                       |    \|    \|
+#                   (1) +-----+-----+(3)
+#                            (2)
+    grid = generate_grid(Triangle, (2, 2));
+    topology = ExclusiveTopology(grid);
+    addfaceset!(grid, "all", x->true)
+    addvertexset!(grid, "all", x->true)
+    @test ∪([Ferrite.getfaceinstances(grid, topology,face) for face in topology.face_skeleton]...)  == grid.facesets["all"]
+    skeletonvec=sort!(collect(grid.facesets["all"]), by = x->x[1]*10+x[2])
+    for faceindex in skeletonvec
+            _replicas = Ferrite.getfaceinstances(grid, topology, faceindex)
+            delete!(_replicas, faceindex)
+            for replica in _replicas
+                deleteat!(skeletonvec, findfirst(x->x==replica, skeletonvec))
+            end 
+    end
+    @test skeletonvec == topology.face_skeleton
```

Comment:
Sorry I am not sure if I understand the test here correctly. What exactly should be covered by this test?

Reply:
Sure. I'll work on them, thanks!
There is a 3D tetrahedron test [here](https://github.com/Ferrite-FEM/Ferrite.jl/pull/606/files#diff-b3cc8751aa66a7415a22b74924d286253575d54f4b024d38d32b03b3fb8fbab9R198-R201):
```julia
# Tetrahedron grid test
    grid = generate_grid(Tetrahedron, (10, 10, 10))
    topology = ExclusiveTopology(grid)
    @test extractboundary(grid, topology) == extractboundarycheck(grid)
```
Regarding this test, it's testing `getfaceinstances` by reconstructing the mesh skeleton using it. If it's working as expected it should return all the instances of the current face -inclusive-, then removing all the instances except the first one from the full faceset "all" should return the face skeleton from my understanding.
I guess there is no need for the `*10+x[2]` part, I'll check it.
Thanks!